pkg/bloomcompactor/controller.go

```diff
@@ -104,6 +113,51 @@ func (s *SimpleBloomController) compactTenant(
 		return errors.Wrap(err, "failed to build gaps")
 	}
 
+	// combine built and superset metas
+	// in preparation for removing outdated ones
+	combined := append(superset, built...)
+
+	outdated := outdatedMetas(combined)
+	level.Debug(logger).Log("msg", "found outdated metas", "outdated", len(outdated))
+	for _, meta := range outdated {
+		for _, block := range meta.Blocks {
+			err := client.DeleteBlocks(ctx, []bloomshipper.BlockRef{block})
+			if err != nil {
+				if client.IsObjectNotFoundErr(err) {
+					level.Debug(logger).Log("msg", "block not found while attempting delete, continuing", "block", block.String())
+				} else {
+					level.Error(logger).Log("msg", "failed to delete block", "err", err, "block", block.String())
+					return errors.Wrap(err, "failed to delete block")
+				}
+			}
+			level.Debug(logger).Log("msg", "removed outdated block", "block", block.String())
```

Comment:
I think it would be nice to have a metric to track how many blocks/metas are deleted